kuksa-client/kuksa_client/grpc/__init__.py

```diff
@@ -917,54 +1209,104 @@ def get(self, entries: Iterable[EntryRequest], **rpc_kwargs) -> List[DataEntry]:
+            rpc_kwargs.get("metadata")
+        )
         req = self._prepare_get_request(entries)
         try:
-            resp = self.client_stub.Get(req, **rpc_kwargs)
+            resp = self.client_stub_v1.Get(req, **rpc_kwargs)
         except RpcError as exc:
             raise VSSClientError.from_grpc_error(exc) from exc
         return self._process_get_response(resp)
 
     @check_connected
-    def set(self, updates: Collection[EntryUpdate], **rpc_kwargs) -> None:
+    def set(
+        self, updates: Collection[EntryUpdate], v1: bool = True, **rpc_kwargs
+    ) -> None:
         """
         Parameters:
             rpc_kwargs
                 grpc.*MultiCallable kwargs e.g. timeout, metadata, credentials.
         """
         rpc_kwargs["metadata"] = self.generate_metadata_header(
-            rpc_kwargs.get("metadata"))
+            rpc_kwargs.get("metadata")
+        )
         paths_with_required_type = self._get_paths_with_required_type(updates)
         paths_without_type = [
-            path for path, data_type in paths_with_required_type.items() if data_type is DataType.UNSPECIFIED
+            path
+            for path, data_type in paths_with_required_type.items()
+            if data_type is DataType.UNSPECIFIED
         ]
         paths_with_required_type.update(
-            self.get_value_types(paths_without_type, **rpc_kwargs))
-        req = self._prepare_set_request(updates, paths_with_required_type)
-        try:
-            resp = self.client_stub.Set(req, **rpc_kwargs)
-        except RpcError as exc:
-            raise VSSClientError.from_grpc_error(exc) from exc
-        self._process_set_response(resp)
+            self.get_value_types(paths_without_type, **rpc_kwargs)
+        )
+        if v1:
+            req = self._prepare_set_request(updates, paths_with_required_type)
+            try:
+                resp = self.client_stub_v1.Set(req, **rpc_kwargs)
+            except RpcError as exc:
+                raise VSSClientError.from_grpc_error(exc) from exc
+            self._process_set_response(resp)
+        else:
+            logger.info("Using v2")
+            if len(updates) == 0:
+                raise VSSClientError(
+                    error={
+                        "code": grpc.StatusCode.INVALID_ARGUMENT.value[0],
+                        "reason": grpc.StatusCode.INVALID_ARGUMENT.value[1],
+                        "message": "No datapoints requested",
+                    },
+                    errors=[],
+                )
+            for update in updates:
+                req = self._prepare_publish_value_request(
+                    update, paths_with_required_type
+                )
+                try:
+                    resp = self.client_stub_v2.PublishValueRequest(req, **rpc_kwargs)
+                except RpcError as exc:
+                    if exc.code() == grpc.StatusCode.UNIMPLEMENTED:
+                        self.set(updates)
```

Comment:
I am thinking for debug purposes if it makes sense to add a debug here as well "V2 not implemented, fallback to V1", so that it is clear if you run in debug mode that v2 actually wasn't supported